lib/practice_exercises.rb

```diff
@@ -1,13 +1,42 @@
 
-# Time Complexity: ?
-# Space Complexity: ?
+# Time Complexity: O(n)
+# Space Complexity: O(1) - I don't feel confident about this though because of the creation of the comparison array, which I think would double the memory needed.
 def is_palindrome(string)
-  raise NotImplementedError, "Not implemented yet"
+  string = string.downcase.gsub(/[^A-Za-z]/, "").split("")
+  comparison = []
+  i = 0
+  j = -1
+
+  until string[i] == string[j] do
+    comparison << (string[i] == string[j])
+    i += 1
+    j -= 1
+  end
+
+  return comparison.include?(false) == false
 end
 
-# Time Complexity: ?
-# Space Complexity: ?
+# Time Complexity: O(n)
+# Space Complexity: O(1)
 def longest_prefix(strings)
```

Comment:
I would say the time complexity is O(n * m) where m is the length of the shortest word and n being the number of strings.  Otherwise 👍